projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -226,4 +289,223 @@ public function get_items( $request ) {
+			}
+
+			$result = Connections::wpcom_create_connection( $input );
+
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+
+			$connection = Connections::get_by_id( $result );
+			l( '>>>>>>>> CONNECTION CREATED', $connection );
+
+			$response = $this->prepare_item_for_response( $connection, $request );
+			$response = rest_ensure_response( $response );
+
+			$response->set_status( 201 );
+
+			return $response;
+
+		}
+
+		$proxy = new Proxy_Requests( $this->rest_base );
+
+		$response = $proxy->proxy_request_to_wpcom( $request );
+
+		l( '>>>>>>>> CREATE CONNECTION RESPONSE', $response );
+
+		if ( is_wp_error( $response ) ) {
+			return new WP_Error(
+				'jp_connection_update_failed',
+				__( 'Something went wrong while creating a connection.', 'jetpack-publicize-pkg' ),
+				$response->get_error_message()
+			);
+		}
+
+		$response = rest_ensure_response( $response );
+
+		$response->set_status( 201 );
+
+		return $response;
+	}
+
+	/**
+	 * Checks if a given request has access to update a connection.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to create items, WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		$permissions = parent::publicize_permissions_check();
+
+		if ( is_wp_error( $permissions ) || ! $permissions ) {
+			return $permissions;
+		}
+
+		// If the user cannot manage the connection, they can't update it either.
+		if ( ! $this->manage_connection_permission_check( $request ) ) {
+			return false;
+		}
+
+		// If the connection is being marked/unmarked as shared.
+		if ( $request->has_param( 'shared' ) ) {
+			// Only editors and above can mark a connection as shared.
+			return current_user_can( 'edit_others_posts' );
+		}
+
+		return current_user_can( 'publish_posts' );
+	}
+
+	/**
+	 * Update a connection.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$connection_id = $request->get_param( 'connection_id' );
+
+		if ( Publicize_Utils::is_wpcom() ) {
+
+			$input = array(
+				'shared' => $request->get_param( 'shared' ),
+			);
+
+			$result = Connections::wpcom_update_connection( $connection_id, $input );
+
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+
+			$connection = Connections::get_by_id( $connection_id );
+			l( '>>>>>>>> CONNECTION UPDATED', $connection );
+
+			$response = $this->prepare_item_for_response( $connection, $request );
+			$response = rest_ensure_response( $response );
+
+			$response->set_status( 201 );
+
+			return $response;
+		}
+
+		$proxy = new Proxy_Requests( $this->rest_base );
+
+		l( '>>>>>>>> UPDATE CONNECTION REQUEST', $request );
+
+		$response = $proxy->proxy_request_to_wpcom( $request, $connection_id );
```

Comment:
I got it to work by passing the connection ID as the path here, otherwise it tried to create a connection.

Reply:
Ah great. That is what that $path is for. I thought that `$request` object will handle that itself.

Thank you for looking into it.

Reply:
Yeah I think the parameter is in the request object, but it didn't add it to the URL when passing the request to WPCOM.